### PR TITLE
Delete probably-unused TrackingFields.tpl

### PR DESCRIPTION
**Pull request by colemanw**
Upgrade Note
----------------------------------------
If you've landed here from a CiviCRM upgrade and are wondering what action is needed, the answer is "probably nothing." 

This PR removes the `TrackingFields.tpl` Smarty template, which is not used by CiviCRM core or any known CiviCRM extensions. In theory it could have been utilized by custom (in-house) code to hide certain custom fields from message templates.

If your site doesn't contain any custom code (e.g. in-house modules or extensions that were written just for you), then you can stop reading now.

If you think your site *might* have custom code that does something with the display of custom fields in messages, you can double-check by searching your code for the text "trackingFields" - if that turns up nothing, you can stop reading now.

If you *did* find something, read on, and feel free to reach out to discuss on https://chat.civicrm.org/civicrm/channels/dev - we'll all be very surprised!

Details
--------------
This PR removes a template and a bunch of conditional logic added 14 years ago to support a "CiviTracker" feature that was never implemented in core and cannot be found in Universe.

The template `TrackingFields.tpl` was throwing e-notices all over the event registration page so I tried to find where the undefined variable `$trackingFields` was supposed to be assigned. I couldn't find it in core, couldn't find it anywhere in Universe, so then I wondered why the template exists at all.

Some digging turned up this issue: https://issues.civicrm.org/jira/browse/CRM-4493
And this svn commit: https://github.com/civicrm/civicrm-svn/commit/4ea717
And this blog post: [https://civicrm.org/blog/lobo/tracking-contributions-and-event-registrations-for-campaigns-and-appeals](http://civicrm.org/node/562)

The blog post proposes end users should write their own "CiviTracker" module in-house that would take advantage of this template. However, 14 years later no "CiviTracker" extension was ever published, and if it were made today there would be better ways of accomplishing the tracking, so the template and conditionals still wouldn't be needed.

Given that the last reference to it is [some obsolete Drupal6 example code](https://github.com/civicrm/civicrm-drupal/blob/299ebc467468a58493f6a763d82df6a9bc744fe8/civitest.module.sample#L617-L626), I doubt anyone is using this template (or even knows about it).

If by some chance someone *has* created their own in-house Drupal 6 module *and* maintained it for all these years *and* ported it to whatever platform they migrated to from D6, this removal wouldn't completely break their functionality, it would just fail to hide the tracker field from their profile. Alternately, they could update their custom fields to use the new `Type=Hidden` which should give very similar functionality.

Note: This also includes a lot of conditionals in the message templates. Using metadata to suppress hidden fields from message templates is a different ball-o-wax, which we've started discussing here https://github.com/civicrm/civicrm-core/pull/27235; but I don't think we need that problem solved in order to merge this.

### CODE REVIEW

File: xml/templates/message_templates/contribution_online_receipt_html.tpl [hunk 1]
```diff
@@ -418,7 +417,6 @@
         </tr>
        {/if}
```
Comment (larssandergreen) [inline xml/templates/message_templates/contribution_online_receipt_html.tpl:418]: It's this one that needs to go, not the one two lines down.

**Comment by larssandergreen** [comment]
Glad you dug in on this one. I started looking once and wasn't really sure it was unused, so left it.

I guess someone could in theory be using this, though it is unlikely, so I think merging this and letting them complain seems reasonable.

I think fixing the one little error above should resolve the test fails and then I think this is good to go once passing.

**Comment by larssandergreen** [comment]
Though I'll also say - even 14 years later, this is definitely still an obvious feature gap. Would be nice to implement something that handle UTM params.

**Comment by demeritcowboy** [comment]
Related, and mostly confirms the same thoughts: https://github.com/civicrm/civicrm-core/pull/25614

**Comment by colemanw** [comment]
As @MegaphoneJon pointed out in https://github.com/civicrm/civicrm-core/pull/25614#discussion_r1117825901 the obviously-better way of doing this would be to implement custom fields of type "Hidden". That's how this feature ought to have been done in the first place IMO.

**Comment by eileenmcnaughton** [comment]
@colemanw my feeling is this is OK to merge as long as we communicate it somehow - possibly in an upgrade message that links to a change log

**Comment by colemanw** [comment]
@eileenmcnaughton ok I have updated this PR to set a Pre-Upgrade message:

image

**Comment by eileenmcnaughton** [comment]
@colemanw OK - I agree that maintaining these fields has been a nightmare & we have no evidence they are used

**Comment by colemanw** [comment]
@eileenmcnaughton sorry, juggling too many branches, I forgot to actually push the commit. Here: #27276

**Comment by composerjk** [comment]
Was the intention to remove all references to `trackingFields`? If so, there are still some in those `.tpl` files. For example:
https://github.com/civicrm/civicrm-core/blob/5159715422a59556294d10cc056b5a335a6cf9ba/xml/templates/message_templates/membership_online_receipt_html.tpl#L533

Looks like some of the ones around `customPost` were missed. fyi.

**Comment by eileenmcnaughton** [comment]
@composerjk yeah - I guess those were missed

**Comment by agileware-justin** [comment]
So here I am wondering what needs to be **done** in response to this pre-upgrade message - which links to this Github PR.
I expect many CiviCRM users will be asking the same question.

> If your site uses custom code to inject tracking fields into messages, it may need updating. See this issue for details, https://github.com/civicrm/civicrm-core/pull/27233

Upgrade messages should provide a clear instructions as to what actions need to be taken, unambiguous.

**Comment by colemanw** [comment]
@agileware-justin ok I've updated the description. Let me know if that's clearer.

**Comment by agileware-justin** [comment]
Thanks @colemanw maybe change heading "Overview" to "Upgrade Instructions"

**Pull request merged**